sopel/tests/factories.py

```diff
@@ -75,22 +99,44 @@ def __call__(self, settings: config.Config) -> bot.Sopel:
 class ConfigFactory:
     """Factory to create settings.
 
+    :param tmpdir: a test folder to store documentation files
+
+    An instance of this factory can be used as a callable to create an instance
+    of :class:`~sopel.config.Config`.
+
+    .. versionchanged:: 8.1
+
+        This factory used to depend on pytest's fixture ``tmpdir``, but is now
+        using Python's standard :class:`pathlib.Path` instead.
+
     .. seealso::
 
         The :func:`~sopel.tests.pytest_plugin.configfactory` fixture can be
         used to instantiate this factory.
     """
-    def __init__(self, tmpdir):
-        self.tmpdir = tmpdir
+    def __init__(self, tmpdir: pathlib.Path) -> None:
+        self.tmpdir: pathlib.Path = tmpdir
```

Comment:
Doubling up on the annotation looks a little weird to me, was `mypy` complaining?

Reply:
Sphinx was! For some reason `from pathlib import Path` confuses Sphinx that cannot find the reference. 😓